changelog.md

```diff
@@ -1,5 +1,9 @@
 # Changelog
 
+## NEXT
+
+- Fixed a bug to improve checking for cycles when adding edges by avoiding repeated checks.
```

Comment:
"Fixed a bug" implies that the behaviour was wrong, people might think we were allowing cycles before this in some cases. Would just say "Improved performance of checking for cycles..."